### PR TITLE
Add editorconfig

### LINKED ISSUE #13: Add editorconfig file
Add **. editorconfig**  file

### PR DESCRIPTION
**Pull request by redxtech**
closes #13

### CODE REVIEW

**Comment by Hsankesara** [comment]
hey, @redxtech  can you please tell me what .idea/vcs.xml do...

**Comment by IOAyman** [comment]
@Hsankesara, it was probably committed by mistake.

**Comment by redxtech** [comment]
@Hsankesara it was added by my IDE, and wasn't shown in the IDE, so I didn't notice it. It has been removed. Do you have .gitignore generator @IOAyman ?

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,9 @@
+root = true
+
+[*]
+charset = utf-8
+indent_style = space
+indent_size = 4
+end_of_line = lf
+insert_final_newline = true
+trim_trailing_whitespace = true
```
Comment (IOAyman) [inline .editorconfig:9]: It is better to turn off trimming for `.md` files. It could cause unwanted behavior.

**Review by IOAyman — CHANGES_REQUESTED** [review]

**Comment by IOAyman** [comment]
@redxtech, Yes I do.  I use the most popular one in the WebStorm plugins repository.

Repo: https://github.com/hsz/idea-gitignore

File: .editorconfig (hunk 1 above)
Comment (IOAyman) [reply .editorconfig:9]:
```
[*.md]
trim_trailing_whitespace = false
```

**Comment by redxtech** [comment]
@IOAyman I use that one too, it's definitely the best I've come across.

Pushed changes to fork to remove the whitespace trim on .md files.

**Review by IOAyman — APPROVED** [review]
LGTM 👍

**Comment by redxtech** [comment]
ignore the typo in the commit message :/

**Comment by Hsankesara** [comment]
looks good to me 👍

**Review by shivamsaboo17 — APPROVED** [review]

**Pull request merged**